request_test.go

```diff
@@ -517,3 +517,67 @@ func TestExecuteRequestSetters(t *testing.T) {
 		Args(args)
 	assertBodyEqual(t, refBuf.Bytes(), req)
 }
+
+func TestPrepareRequestSetters(t *testing.T) {
+	var refBuf bytes.Buffer
+
+	refEnc := msgpack.NewEncoder(&refBuf)
+	err := RefImplPrepareBody(refEnc, validExpr)
+	if err != nil {
+		t.Errorf("An unexpected RefImplPrepareBody() error: %q", err.Error())
+		return
+	}
+
+	req := NewPrepareRequest(validExpr)
+	assertBodyEqual(t, refBuf.Bytes(), req)
+}
+
+func TestUnprepareRequestSetters(t *testing.T) {
+	var refBuf bytes.Buffer
+
+	stmt := PreparedStatement{}
+
+	refEnc := msgpack.NewEncoder(&refBuf)
+	err := RefImplUnprepareBody(refEnc, stmt)
+	if err != nil {
+		t.Errorf("An unexpected RefImplUnprepareBody() error: %q", err.Error())
+		return
+	}
+
+	req := NewUnprepareRequest(&stmt)
+	assertBodyEqual(t, refBuf.Bytes(), req)
+}
```

Comment:
I do not think so. There is no point in `UnprepareRequest` and `ExecutePreparedRequest` without `Prepared`. Let it be better to be without setters.

Reply:
But is not it pretty comfortable to use a single object for unpreparing different statements?
For example: 
```go
stmt := NewPrepared("stmt1")
stmt2 := NewPrepared("stmt1")
unprepareReq := NewUnprepareReq()

conn.Do(unprepareReq.WithPrepared(stmt))
conn.Do(unprepareReq.WithPrepared(stmt2))
```
Less allocations + comfortability.

Reply:
> But is not it pretty comfortable to use a single object for unpreparing different statements? For example:
> 
> ```go
> stmt := NewPrepared("stmt1")
> stmt2 := NewPrepared("stmt1")
> unprepareReq := NewUnprepareReq()
> 
> conn.Do(unprepareReq.WithPrepared(stmt))
> conn.Do(unprepareReq.WithPrepared(stmt2))
> ```
> 
> Less allocations + comfortability.

We assume that request objects are immutable. That's not technically true, but all usage in examples and tests (should be at least):

1. Constuct a request.
2. Send the request to a Connection.

If we assume that request may be immutable this will make life very difficult. We will need additional allocations (copy of a request object at least) in Connection.Do().